### PR TITLE
Add drop oindex after create oindex sql case

### PR DESCRIPTION
**Pull request by LuciferYang**
## What changes were proposed in this pull request?
Add `drop oindex` sql after `create oindex` sql test case to keep the isolation between case.

## How was this patch tested?
mvn test pass

### CODE REVIEW

File: src/test/scala/org/apache/spark/sql/execution/datasources/oap/DataSourceMetaSuite.scala [hunk 1]
```diff
@@ -239,6 +244,9 @@ class DataSourceMetaSuite extends SharedOapContext with BeforeAndAfter {
     assert(fileHeader2.recordCount === 100)
     assert(fileHeader2.dataFileCount === 3)
     assert(fileHeader2.indexCount === 2)
+    // clean index after case
+    sql("drop oindex index1 on oapt1")
```
Comment (adrian-wang) [inline src/test/scala/org/apache/spark/sql/execution/datasources/oap/DataSourceMetaSuite.scala]: if a case failed before run into this line, the index could still be there.

Comment (adrian-wang) [reply src/test/scala/org/apache/spark/sql/execution/datasources/oap/DataSourceMetaSuite.scala]: something like
```
  withIndex(...) {
    ...
  }
```

**Comment by gczsjdy** [comment]
Can we instead add something to `afterEach` like cleaning the temp dir?

**Comment by linhong-intel** [comment]
@LuciferYang any update on this PR?
I think the `withIndex` is a very good idea.

**Comment by LuciferYang** [comment]
@linhong-intel  doing this change ~

**Comment by LuciferYang** [comment]
@adrian-wang @linhong-intel OapPlannerSuite\HiveOapIndexDDLSuite\OapQuerySuite not extends SharedOapContext, use try finally to drop index, other test suite use withIndex pattern to run drop index sql.

File: src/test/scala/org/apache/spark/sql/test/oap/SharedOapContext.scala [hunk 2]
```diff
@@ -88,4 +88,25 @@ trait SharedOapContextBase extends SharedSQLContext {
         ret ++ set.getOrElse(Nil)
       }
   }
+
+  protected def withIndex(indices: TestIndex*)(f: => Unit): Unit = {
+    try f finally {
+      indices.foreach { index =>
+        val baseSql = s"DROP OINDEX ${index.indexName} on ${index.tableName}"
+        index.partitions.length match {
```
Comment (adrian-wang) [inline src/test/scala/org/apache/spark/sql/test/oap/SharedOapContext.scala]: what about use
`baseSql + index.partitions.map(p => s"${p.key} = '${p.value}'").mkString(" partition (", ",", ")")`

Comment (adrian-wang) [reply src/test/scala/org/apache/spark/sql/test/oap/SharedOapContext.scala]: just ignore this....

Comment (adrian-wang) [inline src/test/scala/org/apache/spark/sql/test/oap/SharedOapContext.scala]: maybe use `if (index.partitions.isEmpty) ... else ...` looks better.

Comment (LuciferYang) [reply src/test/scala/org/apache/spark/sql/test/oap/SharedOapContext.scala]: ok~

**Comment by LuciferYang** [comment]
@adrian-wang thx for ur review ~

**Pull request merged**